app/src/main/java/com/example/playlistmaker/SearchItemAdapter.kt

```diff
@@ -1,26 +1,26 @@
 package com.example.playlistmaker
 
-import android.content.Context.MODE_PRIVATE
-import android.content.Intent
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.ImageView
 import android.widget.TextView
-import androidx.core.content.ContextCompat.startActivity
 import androidx.recyclerview.widget.RecyclerView.Adapter
 import androidx.recyclerview.widget.RecyclerView.ViewHolder
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.resource.bitmap.RoundedCorners
-import com.example.playlistmaker.data.SearchHistory
 import com.example.playlistmaker.data.api.model.Track
 import java.text.SimpleDateFormat
 import java.util.Locale
 
 
-class SearchItemAdapter(private val dataSet: MutableList<Track>) :
+class SearchItemAdapter(
+    private val dataSet: MutableList<Track>,
+    private val onClick: (Int) -> Unit
```

Comment:
```suggestion
    private val onClick: (Track) -> Unit
```
Ты можешь сразу трэк в лямбду отдать, чтобы в активити не доставать по индексу, т.к. при вызове лямбды адаптер знает, на какой трек было нажатие